### PR TITLE
AWS SSM connection: Add ability to specify S3 addressing style

### PR DESCRIPTION
**Pull request by timburnet-systematica**
This is useful for newly created buckets in combination with private s3 endpoints.

##### SUMMARY
This is useful for newly created buckets in combination with private s3 endpoints where the current code just breaks

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
aws_ssm

### CODE REVIEW

**Comment by gillg** [comment]
@markuman possible to merge this PR with https://github.com/ansible-collections/community.aws/pull/854 ?
@james-masson can you add a changlog fragment, version add 3.0.0 in the doc part, and add possible values like "auto" in the doc part ?

File: plugins/connection/aws_ssm.py [hunk 1]
```diff
@@ -64,6 +64,11 @@
     type: integer
     vars:
     - name: ansible_aws_ssm_retries
+  s3_addressing_style:
+    description: Style of S3 URL to use.
+    default: 'auto'
+    vars:
+    - name: ansible_aws_ssm_s3_addressing_style
```
Comment (gillg) [inline plugins/connection/aws_ssm.py:71]: Missing 
```yaml
version_added: 3.0.0
choices:
  - auto
  - virtual
```

**Review by gillg — CHANGES_REQUESTED** [review]

**Comment by Hokwang** [comment]
please take care this PR. I think #705 should not close yet.

**Comment by jatorcasso** [comment]
recheck

**Comment by adsanz** [comment]
Note: this issue fixes #637 (which appears closed, but actually this PR is the fix) I've implemented the PR on my collection install and test it against a newly created bucket (less than 1 month old) which had the same issue as described by #637 and it worked. Is there any chance this gets merged soon? As anybody with a newly created bucket for SSM connections is probably going to face the same problem.

**Comment by tremble** [comment]
Thanks for taking the time to submit this PR.  It appears to have stalled out, and because I can't push to the branch I can't revive this.

The work has been revived in #1633 and will hopefully be available in 5.2.0 (Real Soon Now, honest)

**Pull request closed without merging**